commit 979b5671a61f
Author: Ryan Hunt <rhunt@eqrion.net>
Date:   Mon Oct 9 12:49:53 2017 -0500

    Bug 1407001 - Don't reset compositors for composited changed callback when GPUProcessManager is null. r=lsalzman
    
    MozReview-Commit-ID: 4E31bt3Rzkz
    
    --HG--
    extra : rebase_source : 84a18ffcc77f37220284679c1dbb916a25524d31
---
 widget/gtk/nsWindow.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git widget/gtk/nsWindow.cpp widget/gtk/nsWindow.cpp
index 575dc83d964c..eb2f0e5e95cd 100644
--- widget/gtk/nsWindow.cpp
+++ widget/gtk/nsWindow.cpp
@@ -5923,7 +5923,13 @@ check_resize_cb (GtkContainer* container, gpointer user_data)
 static void
 screen_composited_changed_cb (GdkScreen* screen, gpointer user_data)
 {
-    GPUProcessManager::Get()->ResetCompositors();
+    // This callback can run before gfxPlatform::Init() in rare
+    // cases involving the profile manager. When this happens,
+    // we have no reason to reset any compositors as graphics
+    // hasn't been initialized yet.
+    if (GPUProcessManager::Get()) {
+        GPUProcessManager::Get()->ResetCompositors();
+    }
 }
 
 static void
